browser/src/control/Control.JSDialogBuilder.js

```diff
@@ -2313,19 +2311,19 @@ L.Control.JSDialogBuilder = L.Control.extend({
 
 			if (data.w2icon) {
 				// FIXME: DEPRECATED, this is legacy way to setup icon based on CSS class
-				var buttonImage = L.DomUtil.create('div', 'w2ui-icon ' + data.w2icon, button);
+				buttonImage = L.DomUtil.create('div', 'w2ui-icon ' + data.w2icon);
 			}
 			else if (hasImage !== false){
 				if (data.icon) {
-					buttonImage = L.DomUtil.create('img', '', button);
+					buttonImage = L.DomUtil.create('img', '');
 					this._isStringCloseToURL(data.icon) ? buttonImage.src = data.icon : L.LOUtil.setImage(buttonImage, data.icon, builder.map);
 				}
 				else if (data.image) {
-					buttonImage = L.DomUtil.create('img', '', button);
+					buttonImage = L.DomUtil.create('img', '');
```

Comment:
how does it work as new html node is not added to the DOM?

Reply:
using appendChild at the end of the function